cmake/Modules/KatanaBuildCommon.cmake

```diff
@@ -252,19 +252,19 @@ if (NOT MySQL_FOUND)
   message(STATUS "Library MySQL not found, not building MySQL support")
 endif ()
 
-foreach (llvm_step RANGE 2)
-  # Range is [0, end]. Start version search from highest compatible version
-  # first.
-  math(EXPR llvm_ver "11 - ${llvm_step}")
+# Search for the highest compatible version of LLVM from the list of versions here.
+# LLVM minor versions are not ABI compatible, so we need to specifically say we support the 11.1 special release.
+message(CHECK_START "Looking for LLVM")
+foreach (llvm_ver IN ITEMS 11.1 11.0 10 9)
```

Comment:
I'm wondering if we should prioritize changing how we do command line parsing.

The only reason we depend on libllvm is for the command line parsing. And the only reason I added this check was to check against situations where clang would be installed (as a compiler) but its development libraries were too old.

https://github.com/KatanaGraph/katana-enterprise/issues/834

Anyway, can we keep the compatibility at LLVM 7 and up for now?

Reply:
We didn't support 7 before. We actually only supported 9 and up and no one noticed. I suspect someone updated the max version without increasing the range. That's one reason I moved to using an explicit version list. I was trying to keep the exact same semantics even if they were wrong w.r.t. to the documentation. I will include 7 and 8 to have wider compatibility for the moment.

Re: command line parsing. Yes, we should get ride of the llvm lib dependency. Maybe we could use `boost::program_options` (https://www.boost.org/doc/libs/1_75_0/doc/html/program_options/tutorial.html).